src/app/common/components/MustGatherBtn.tsx

```diff
@@ -0,0 +1,105 @@
+import { getMustGatherApiUrl } from '@app/queries/helpers';
+import { IMustGatherResponse } from '@app/client/types';
+import { UseQueryResult } from 'react-query';
+import { IPlan } from '@app/queries/types';
+
+interface IMustGatherBtn {
+  mg: IMustGatherResponse | undefined;
+  mustGathers: UseQueryResult<IMustGatherResponse[], Response>;
+  customName: string;
+  type: 'plan' | 'vm';
+}
+
+export const MustGatherBtn: React.FunctionComponent<IMustGatherBtn> = ({
+  mg,
+  mustGathers,
+  customName,
+  type,
+}) => {
+  const {
+    setMustGatherModalOpen,
+    setActiveMustGather,
+    mustGatherWatchList,
+    setMustGatherWatchList,
+  } = React.useContext(MustGatherContext);
+
+  return (
+    <>
+      {mg?.status === 'completed' && mg?.['archive-name'] ? (
+        <Tooltip
+          content={
+            !mustGathers.isSuccess
+              ? `Could not reach must gather service.`
+              : `${mg?.['archive-name']} available for download`
+          }
+        >
+          <Button
+            isAriaDisabled={!mustGathers.isSuccess}
+            variant="secondary"
+            component={'a'}
+            target="_blank"
+            download
+            href={getMustGatherApiUrl(`must-gather/${mg?.['id']}/data`)}
+          >
+            Download logs
+          </Button>
+        </Tooltip>
+      ) : (
+        <>
+          <Tooltip
+            content={
+              !mustGathers.isSuccess
+                ? `Could not reach must gather service.`
+                : mg?.status === 'inprogress'
+                ? `Collecting ${type === 'plan' ? 'migration plan' : 'VM migration'} logs.`
+                : mg?.status === 'new'
+                ? `Must gather queued for execution.`
+                : mg?.status === 'error'
+                ? `Could not complete must gather for ${mg['custom-name']}`
+                : `Collects the current ${
+                    type === 'plan' ? 'migration plan' : 'VM migration'
+                  } logs and creates a tar archive file for download.`
+            }
+          >
+            <Button
+              icon={mg?.status === 'error' ? <WarningTriangleIcon /> : null}
+              isLoading={
+                mustGatherWatchList.map((mg) => mg.name).includes(customName) &&
+                !mustGathers.isError &&
+                (mg?.status === 'inprogress' || mg?.status === 'new')
+              }
+              isAriaDisabled={
+                (mustGatherWatchList.map((mg) => mg.name).includes(customName) &&
+                  (mg?.status === 'inprogress' || mg?.status === 'new')) ||
+                !mustGathers.isSuccess
+              }
+              variant="secondary"
+              onClick={() => {
+                setMustGatherModalOpen(true);
+                setActiveMustGather({
+                  type,
+                  customName: customName,
+                  status: 'new',
+                });
+
+                setMustGatherWatchList([
```

Comment:
I think maybe we should add a code comment here describing what you explained to me yesterday about why we need to `setMustGatherWatchList` here instead of relying on the query. We might also want to put a comment in the context provider where the `[mustGatherWatchList, setMustGatherWatchList]` state is defined, because without seeing this usage, it looks like unnecessary duplication of state when you look at the provider code alone.

It might also be worth opening an issue and following up if we can get the backend to reflect this state properly when we reload the list, so we can clean this part up.